polygon/bridge/db.go

```diff
@@ -271,22 +294,21 @@ func (s *MdbxStore) PruneEventIDs(ctx context.Context, blockNum uint64) error {
 	kByte := make([]byte, 8)
 	binary.BigEndian.PutUint64(kByte, blockNum)
 
-	it, err := tx.RangeDescend(kv.BorEvents, nil, kByte, 0)
+	cursor, err := tx.Cursor(kv.BorEventNums)
 	if err != nil {
 		return err
 	}
+	defer cursor.Close()
 
-	for it.HasNext() {
-		k, _, err := it.Next()
+	for k, _, err := cursor.Seek(kByte); k != nil; k, _, err = cursor.Next() {
```

Comment:
if err is != nil and k is == nil - we would miss returning the err
consider something like:

```
	var k []byte
	for k, _, err = cursor.Seek(kByte); err == nil && k != nil; k, _, err = cursor.Next() {
		if err := tx.Delete(kv.BorEventNums, k); err != nil {
			return err
		}
	}
	if err != nil {
		return nil
	}
```